qiime2/sdk/usage.py

```diff
@@ -253,9 +253,10 @@ def _assert_has_line_matching_(self, ref, label, path, expression):
     def get_result(self, ref):
         return self._get_record(ref)
 
-    def _add_outputs_to_scope(self, outputs, derived_outputs):
-        outputs.validate_derived(derived_outputs)
-        for ref, result in derived_outputs.items():
+    def _add_outputs_to_scope(self, outputs, computed_outputs):
+        outputs.validate_computed(computed_outputs)
```

Comment:
> Will this ever actually raise `ValueError`?

It could if an implementation of a Usage driver did it wrong.

The check is there for SDK/Interface developers - in this case folks like you, implementing a Usage driver.

If you mess up the return from `_action_`, this guard might help you get set on the right course.